mitosheet/mitosheet/tests/public/v3/sheet_functions/test_avg.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GPL License.
+"""
+Contains tests for the SUM function.
+"""
+
+import pytest
+import pandas as pd
+import numpy as np
+from mitosheet.public.v3.rolling_range import RollingRange
+
+from mitosheet.public.v3.sheet_functions.number_functions import AVG
+
+
+AVG_VALID_TESTS = [
+    # Just constant tests
+    ([2, 3], 2.5),
+    ([2.0, 3.0], 2.5),
+    ([2.0, '$3.0'], 2.5),
+
+    # Constants and series
+    ([2, pd.Series([1,2,3])], pd.Series([1.5,2,2.5])),
+    ([2, pd.Series([1,2,3])], pd.Series([1.5,2,2.5])),
+    ([2.0, pd.Series([1,2,3])], pd.Series([1.5,2,2.5])),
+    ([2.0, pd.Series([1,2,None])], pd.Series([1.5,2,None])),
+    ([2.0, pd.Series(['1', '2', '3'])], pd.Series([1.5,2,2.5])),
+    ([2.0, pd.Series(['1.0', '2.0', '3.0'])], pd.Series([1.5,2,2.5])),
+    
+    # Dataframes
+    ([pd.DataFrame({'a': [1, 1, 1], 'b': [2, 2, 2]}), pd.Series([1,2,3])], pd.Series([10/7, 11/7, 12/7])),
+    ([pd.DataFrame({'a': [1, 1, 1], 'b': [2, 2, 2]}), 2], 11/7),
+    ([pd.DataFrame({'a': ['$1', '$1', '$1'], 'b': [2, None, 2]}), pd.Series([1,2,3])], pd.Series([8/6, 9/6, 10/6])),
```

Comment:
NaN values can represent one a few things. With these tests, they are most closely related to cells that are just not numbers. E.g. if you sum a range that has "abc" in them, it will just skip these cells. 

So: NaN is not empty cells only. It's also cells that don't match the type of the function (at least for SUM). And other functions might handle these differently (e.g. I bet not every function skips them). So we'll just do our best here.